spec/controllers/concerns/collaborator_processing_spec.rb

```diff
@@ -115,7 +115,93 @@ class FakesController < ApplicationController
 
-  context 'removing users' do
+  context 'adding collaborator groups' do
+    let!(:group_member) { create(:group_member) }
+    let(:group) { group_member.group }
+
+    let!(:group_member2) { create(:group_member, group: group) }
+
+    context 'adding a single group' do
+      before do
+        expect(cookbook.owner).to eq(fanny)
+        sign_in fanny
+
+        allow(Group).to receive(:find).and_return(group)
+        allow(group).to receive(:members).and_return(group.members)
+      end
+
+      it 'finds the correct group' do
+        expect(Group).to receive(:find).with(group.id.to_s).and_return(group)
+        subject.add_group_members_as_collaborators(cookbook, "#{group.id}")
+      end
+
+      it 'associates the group with the resource' do
+        expect(group).to receive(:group_resources).at_least(:once).and_return([])
+        new_group_resource = create(:group_resource, group: group)
+
+        allow(GroupResource).to receive(:create!).and_return(new_group_resource)
+
+        expect(group.group_resources).to receive(:<<).with(new_group_resource)
+
+        subject.add_group_members_as_collaborators(cookbook, "#{group.id}")
+      end
+
+      it 'finds the members for the group' do
+        expect(group).to receive(:members).and_return(group.members)
+        subject.add_group_members_as_collaborators(cookbook, "#{group.id}")
+      end
+
+      it 'maps the user ids' do
+        expect(group.members).to receive(:map).and_return(group.members.map(&:id))
+        subject.add_group_members_as_collaborators(cookbook, "#{group.id}")
+      end
+
+      it 'transforms the user ids into a string' do
+        member_ids = group.members.map(&:id)
+        allow(group.members).to receive(:map).and_return(member_ids)
+        expect(member_ids).to receive(:map).and_return(member_ids.map(&:to_s))
+
+        subject.add_group_members_as_collaborators(cookbook, "#{group.id}")
+      end
+
+      it 'makes a collaborator for each group user' do
+        user_ids = group.members.map(&:id).map(&:to_s)
+
+        expect(subject).to receive(:add_users_as_collaborators).with(cookbook, user_ids, group.id.to_s)
+        subject.add_group_members_as_collaborators(cookbook, "#{group.id}")
+      end
+
+      it 'associates the collaborator with the group' do
+        expect(cookbook.collaborators).to be_empty
+        subject.add_group_members_as_collaborators(cookbook, "#{group.id}")
+        cookbook.collaborators.each do |collaborator|
+          expect(collaborator.group).to eq(group)
+        end
+      end
+    end
+
+    context 'adding multiple groups' do
+      let!(:group2_member) { create(:group_member) }
+      let(:group2) { group2_member.group }
+      let!(:group2_member2) { create(:group_member, group: group2) }
+
+      it 'finds all groups' do
+        expect(Group).to receive(:find).at_least(:twice).and_return(group, group2)
+        subject.add_group_members_as_collaborators(cookbook, "#{group.id}, #{group2.id}")
+      end
```

Comment:
Good thinking...but I think I'd rather leave it as it is now.  Thanks for the suggestions, though!